trin-core/src/portalnet/storage.rs

```diff
@@ -127,20 +155,28 @@ impl PortalStorage {
     }
 
     /// Public method for storing a given value for a given content-key.
-    pub fn store(&mut self, key: &String, value: &String) -> Result<(), PortalStorageError> {
-        let content_id = self.content_key_to_content_id(key);
-
+    pub async fn store(
+        &mut self,
+        key: &ContentKey,
+        value: &Vec<u8>,
+    ) -> Result<(), PortalStorageError> {
+        let content_id = match key {
+            ContentKey::StateContentKey(val) => val.derive_content_id().unwrap(),
+            ContentKey::HistoryContentKey(val) => val.derive_content_id().unwrap(),
+        };
+        let content_id = Into::<[u8; 32]>::into(content_id);
         let distance_to_content_id = self.distance_to_content_id(&content_id);
 
         // Check whether data is outside our radius.
         if distance_to_content_id > self.data_radius {
-            debug!("Not storing: {}", key);
-            return Ok(());
+            debug!("Not storing: {:02X?}", key);
+            return Err(PortalStorageError::OutsideDistanceError);
         }
 
         // Store the data.
         self.db_insert(&content_id, value)?;
-        self.meta_db_insert(&content_id, &key, value)?;
+        self.meta_db_insert(&content_id, &key.to_bytes(), value)
```

Comment:
It seems that there is nothing that would rollback the `db_insert` if the `meta_db_insert` fails here, which would allow the two to move out of sync. Not sure what the best way to enforce this synchronization is, but we need to have error handling for such cases. Similarly for remove.

Reply:
Hmmmm, I guess we could alway just manually perform the opposite operation on `rocksdb` if `metadb` fails? That seems like the desired behavior right?
Something like...
```rust
self.db_insert(content_id, value)?;
match self.meta_db_insert(....).unwrap() {
    Ok(_) => (),
    Err(_) => self.db_remove(content_id)?,
}
```

Reply:
Cool, the solution I implemented is fairly simple, and based mostly under the assumption that if the `rocksdb` operation succeeded before we reach the `metadb` operation, then the corrective/reverting `rocksdb` operation **should** succeed. Tbf this is not a guarantee, but should be good for now. At this point, I do think the greater issue of enforcing synchronicity between the two databases is not over (including some better testing), but is beyond the scope of this pr and can be tracked in an issue. Lmk what you think